AutoLayout Assignment/Ass2/Ass2/CustomView.swift

```diff
@@ -0,0 +1,191 @@
+        allLabels()
+       buttonInfo()
+        
+    }
+    func addSubviews(){
+        self.addSubview(labelScore)
+        self.addSubview(roundlabel)
+        self.addSubview(hitMe)
+        self.addSubview(labelRandom)
+        self.addSubview(buttonInfoo)
+        self.addSubview(mySlider)
+        self.addSubview(buttonStart)
+    }
+    required init?(coder: NSCoder) {
+            fatalError("init(coder:) has not been implemented")
+        }
+    
+    func allLabels(){
+        labelScore.text = "Score: "
+        roundlabel.text="Round: \(round)"
+        hitMe.text="Hit ME"
+        hitMe.textColor=UIColor.blue
+        hitMe.font = hitMe.font.withSize(25)
+        
+    }
+    func labelRandomm(){
+        labelRandom.textAlignment = .center
+        count = Int.random(in: 1..<100)
+        labelRandom.text = "Try to Hit as close as possible to:   \(count)"
+    }
+    
+    func buttonStartOver(){
+        buttonStart.setTitle("Start Over", for: .normal)
+        buttonStart.setTitleColor(UIColor.red, for: .normal)
+        buttonStart.addTarget(self, action: #selector(startOnClick), for: .touchUpInside)
+        
+    }
+    
+    @objc func startOnClick() {
+        count=0
+        round=0
+        roundlabel.text="Round: \(round)"
+        labelScore.text = "Score: "
+        mySlider.value=0
+        dict=[:]
+    }
+    
+    func buttonInfo(){
+        buttonInfoo.setTitle("Info", for: .normal)
+        buttonInfoo.setTitleColor(UIColor.red, for: .normal)
+        buttonInfoo.addTarget(self, action: #selector(infoOnClick), for: .touchUpInside)
+        
+    }
+    
+    @objc func infoOnClick() {
+        showAlert()
+    }
+    
+    func sliderr(){
+        mySlider.minimumValue = 0
+        mySlider.maximumValue = 100
+        mySlider.isContinuous=false
+        mySlider.addTarget(self, action: #selector(self.sliderValueDidChange), for: .valueChanged)
+        
+    }
+    
+    @objc func sliderValueDidChange(){
+        let sliderValue = Int(mySlider.value)
+        var score = 100 - (2 * abs(sliderValue - count))
+        if(score<0){
+            score=0
+        }
+        
+        round+=1
+        roundlabel.text="Round: \(round)"
+        labelScore.text = "Score: \(score)"
+        count = Int.random(in: 1..<100)
+        labelRandom.text = "Try to Hit as close as possible to:   \(count)"
+        if round != 0{
+            dict[round] = score
+        }
+    }
+    
+      func showAlert() {
+        // customise your view
+        popup = UIView(frame: CGRect(x: 100, y: 200, width: 200, height: 500))
+        popup.backgroundColor = UIColor.red
+        let lb = UILabel(frame: CGRect(x: 30, y: 20, width: 300, height: 450))
+          
+          var txt=""
+          for i in 0..<dict.count{
+              let val = dict[i+1] ?? 0
+              txt = txt + "Round \(i+1) score is \(val) \n"
+              
+          }
+          lb.numberOfLines = dict.count
+          lb.text=txt
```

Comment:
What would happen if label height exceeds hardcoded height? should we use auto layout to prevent it?